src/commands/monitor.rs

```diff
@@ -523,15 +524,22 @@ async fn submit_and_watch_solution<T: MinerConfig + Send + Sync + 'static>(
 	nonce: u64,
 	listen: Listen,
 	dry_run: bool,
+	at: &Header,
 ) -> Result<(), Error> {
 	let tx = epm::signed_solution(RawSolution { solution, score, round })?;
-
-	let xt = client.chain_api().tx().create_signed_with_nonce(
-		&tx,
-		&signer,
-		nonce as u64,
-		Default::default(),
-	)?;
+	let signed_phase_len = client
```

Comment:
NOTE: this is static applies the extrinsic to live for "signed_phase_len" and if for example the last block in the sign phase it will "live" for additional blocks.

We could change that as well but that is a slightly bigger change.

Reply:
Yes, I think this approximation is OK given that we also have the checks on the EPM side. 

otoh, maybe it would be easier to know exactly the boundaries of the signed phase in on terms of block number if the `Phase::Signed` in EPM wrapped the block number when it started? That way the miner could fetch the current EPM phase and know `mortal(signed_phase_started, signed_phase_len)`. is this worth the change?

Reply:
Yeah, indeed it requires some changes because each block is executed in its own async task and needs synchronization.

Lemme create another issue for it :)

Reply:
Thanks! I think for now this and the EPM-side changes are enough to tackle the original issue.